### PR TITLE
add compile flag for number parsing functions

### PR DESCRIPTION
**Pull request by jonasschnelli**
- disabling tests in case of providing number parsing function (only build library)

### CODE REVIEW

**Comment by jgarzik** [comment]
1) I agree with the changes removing numberparse.h

2) Disabling tests is disappointing - and apparently unnecessary?

3) I do not understand the need for a compile time flag.  If numberparse is specific to tests, then move that module to a "libtest.a"

**Comment by jonasschnelli** [comment]
2) My fault. Just removed the exclusion.

3) If i'd like to use UniValue in a non bitcoin-core project, i require the `ParseIntXXX()` functions and therefor i'd like to have these functions within libunivalue.a. In case of bitcoin-core, we wan't to avoid having duplicated symbols... Do you see a better way? Copy the source code and rename/namespace the parsing function?

File: Makefile.am [hunk 1]
```diff
@@ -6,8 +6,14 @@ lib_LIBRARIES = libunivalue.a
 libunivalue_a_SOURCES = \
 	univalue.cpp \
 	univalue_read.cpp \
-	univalue_write.cpp \
-	numberparse.cpp
+	univalue_write.cpp
+
+# ---------------------------------------------------
+# only compile tests if numberparse has been included
+# ---------------------------------------------------
```
Comment (jgarzik) [inline Makefile.am]: Comment is incorrect

**Comment by jgarzik** [comment]
OK.  Looks good once comment is fixed.

File: Makefile.am (hunk 1 above)
Comment (jonasschnelli) [inline Makefile.am]: Right. Fixed.

**Pull request merged**